pandas/core/arrays/masked.py

```diff
@@ -247,11 +247,16 @@ def __setitem__(self, key, value) -> None:
 
     def __iter__(self) -> Iterator:
         if self.ndim == 1:
```

Comment:
What if I set an NAvalue during iteration?

Reply:
I guess a lot of iter methods wouldn’t be robust to that. Not worth worrrying about

Reply:
If you think its worth handling that case, I can remove the `if` block that iterates without the mask. The `else` block is robust to your example and we still get a perf improvement if we were to hit `'else` block for everything:

```
import pandas as pd
import numpy as np

arr_without_na = pd.array(np.arange(10**6), dtype="Int64")

arr_with_na = arr_without_na.copy()
arr_with_na[::2] = pd.NA

def time_iter(arr):
    for v in arr:
        pass

%timeit time_iter(arr_without_na)
%timeit time_iter(arr_with_na)

# without NA's
214 ms ± 13.5 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)     <- main
56.3 ms ± 1.03 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)  <- PR

# with NA's
273 ms ± 27.7 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)     <- main
85.6 ms ± 902 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)   <- PR
```

Reply:
no not worth worrying about